src/test/scala/dev/kamu/cli/utility/DependencyGraphSpec.scala

```diff
@@ -0,0 +1,126 @@
+package dev.kamu.cli.utility
+
+import org.scalatest.{Matchers, FunSuite}
+
+class DependencyGraphSpec extends FunSuite with Matchers {
+  test("Empty graph or isolated vertices") {
+    val deps = List.empty[(Int, Int)]
+    def dependsOn(v: Int) = deps.filter(d => d._1 == v).map(d => d._2)
+    val g = new DependencyGraph(dependsOn)
+    g.resolve(Nil) should be(Nil)
+    g.resolve(List(0)) should be(List(0))
+    g.resolve(List(1)) should be(List(1))
+    g.resolve(List(0, 1)) should be(List(0, 1))
+    g.resolve(List(1, 0)) should be(List(1, 0))
+  }
+
+  test("Self-dep") {
+    // TODO: test case
+    pending // requires circular dependency resolution support
```

Comment:
let's not merge with `pending`s - either implement the feature or remove the test

Reply:
removed the test